windows/src/desktop/setup/SetupStrings.pas

```diff
@@ -63,59 +71,92 @@ interface
-    'Setup has detected that Keyman Desktop 7.x is installed.  Do you want to copy your Keyman Desktop 7.x keyboard configuration into Keyman Desktop 8?'
-    );
+  {ssApplicationTitle}                        '$APPNAME $VERSION Setup',
+  {ssTitle}                                   'Install $APPNAME $VERSION',
+  {ssWelcome_Keyboards}                       'This install includes:'#13#10+'� $APPNAME $VERSION',
+  {ssInstallSuccess}                          '$APPNAME $VERSION has been installed successfully.',
+  {ssCancelQuery}                             'Are you sure you want to cancel the installation of $APPNAME?',
+
+  {ssFreeCaption}                             '$APPNAME $VERSION is free and open source',
+  {ssLicenseLink}                             '&Read the license',
+  {ssInstallOptionsLink}                      'Install &options',
+
+  {ssOkButton}                                'OK',
+  {ssInstallButton}                           '&Install $APPNAME',
+  {ssCancelButton}                            'Cancel',
+  {ssExitButton}                              'E&xit',
+
+  {ssPackageMissing}                          'Package %0:s (%1:s) is missing.  Setup can continue but will not install this package.',
+
+  {ssErrorDownloadingUpdate}                  'Error %0:d downloading update from server',
+  {ssErrorUnableToContactServer}              'Unable to contact server',
+  {ssErrorUnableToContactServerDetailed}      'Unable to contact server, error was: %0:s',
+
+  {ssStatusCheckingInternetExplorer}          'Checking Internet Explorer version',
+  {ssStatusCheckingForUpdates}                'Checking for updates online',
+  {ssStatusInstalling}                        'Installing $APPNAME',
+  {ssStatusInstallingPackage}                 'Installing package %0:s',
+  {ssStatusComplete}                          'Installation Complete',
+
+  {ssQueryUpdateInternetExplorer}             'Internet Explorer 9.0 or later is required to install $APPNAME.  '+
+                                              'Please use Windows Update to update your system before installing $APPNAME.',   // I4470
+
+  {ssQueryUpdateVersion}                      'Version %1:s of $APPNAME has been released and is available for download.  This update is %0:dKB.  '+
+                                              'Do you want to download and install the updated version (recommended)?',
+  {ssQueryUpdatePackage}                      'The package %0:s is already installed.  Do you wish to update it now?',
+
+  {ssQueryRestart}                            'You must restart Windows before Setup can complete.  When you restart Windows, Setup will continue.'+
+                                              #13#10#13#10'Restart now?',
+  {ssErrorUnableToAutomaticallyRestart}       'Windows was not able to be automatically restarted.  You should restart Windows before you try and start Keyman.',
```

Comment:
So we're not generalizing all of the strings?

Reply:
I'm not sure I understand the drift of the question. These strings were added here to make it possible to customise the visible parts of the Setup UI in Keyman 12, for the FirstVoices installer. It's not intended to be a full g11n or generalization step.

Reply:
Drift of question:  `... before you try and start Keyman.'`

I take it this is not a "visible part of the Setup UI" that should be addressed for use with the FirstVoices installer, then?

Reply:
These are default strings that can be overridden by the custom installer. That said, this message is super rare and I don't really care about it that much ;-)